<a href="https://colab.research.google.com/github/clarawajdenbaum/Project-2/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd 

First choice: dataset 1



In [13]:
df = pd.read_csv('/content/drive/MyDrive/Project 2/docs/archive (2).zip')
df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [7]:
df.shape

(5110, 12)

1) Source of data

https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

2) Brief description of data

This data calculate the perdiction of having a stroke via different featurs (gender, age, hypertension, heart disease,... (see df.head())

3) What is the target?

The target is df['stroke']

4) Is this a classification or regression problem?

This is a classification problem.

5) How many features?

There are 10 features.

6) How many rows of data.

The data contains 5110 (but it hasnt been cleaned yet).

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

The challenges could be that I will have to one hot encode a lot of those columns. 



Second choice: dataset 2

In [15]:
df_2 = pd.read_csv('/content/drive/MyDrive/Project 2/docs/archive (7).zip')
df_2.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [16]:
df_2.shape

(918, 12)

1) Source of data

https://www.kaggle.com/datasets/fedesoriano/heart-failure-prediction

2) Brief description of data

This data predicts heart failure.

3) What is the target?

df['Heart Disease]

4) Is this a classification or regression problem?

classification

5) How many features?

11

6) How many rows of data.

918

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

Again, I will have to one hot encode a lot of features. 